pkg-manager/resolve-dependencies/src/index.ts

```diff
@@ -281,6 +280,21 @@ export async function resolveDependencies (
     }
   }))
 
+  let updatedCatalogs: Record<string, Record<string, string>> | undefined
+  for (const project of projectsToResolve) {
+    if (!project.updatePackageManifest) continue
+    const resolvedImporter = resolvedImporters[project.id]
+    for (let i = 0; i < resolvedImporter.directDependencies.length; i++) {
+      if (project.wantedDependencies[i]?.updateSpec == null) continue
```

Comment:
Interesting, so each index in `directDependencies` would correspond to each index in `wantedDependencies`? Do we really have this guarantee? I fear future changes may break this assumption and introduce hard-to-debug regression.

Reply:
I agree this is very brittle. I should have commented earlier that this was taken from how we update the `package.json` today:

https://github.com/pnpm/pnpm/blob/8a9f3a48357d2b32fbe0c7ac918145f437f6c160/pkg-manager/resolve-dependencies/src/updateProjectManifest.ts#L20-L1

I thought it made sense to carry over the same logic for updating `pnpm-workspace.yaml`, but it's probably worth some kind of refactor of both of these code paths since a break here would be subtle and hard to debug.